File: result/metagpt/claude/humaneval/115.c
#include <stdio.h>

int max_fill(int** grid, int gridSize, int* gridColSize, int capacity) {
    int out = 0;
    for (int i = 0; i < gridSize; i++) {
        int sum = 0;
        for (int j = 0; j < gridColSize[i]; j++) {
            sum += grid[i][j];
        }
        if (sum > 0) {
            out += (sum - 1) / capacity + 1;
        }
    }
    return out;
}

File: result/metagpt/deepseek/humaneval/115.rs
pub fn max_fill(grid: Vec<Vec<i32>>, capacity: i32) -> i32 {
    grid.iter()
        .map(|row| {
            let sum: i32 = row.iter().sum();
            if sum > 0 {
                (sum + capacity - 1) / capacity
            } else {
                0
            }
        })
        .sum()
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_max_fill() {
        assert_eq!(max_fill(vec![vec![1, 2], vec![3, 4]], 2), 4);
        assert_eq!(max_fill(vec![vec![0, 0], vec![0, 0]], 2), 0);
        assert_eq!(max_fill(vec![vec![5]], 3), 2);
    }
}